pkg/vz/vm_darwin.go

```diff
@@ -357,19 +357,19 @@ func attachDisks(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfigura
 		}
 		if extraDiskFormat != "raw" {
 			rawPath := fmt.Sprintf("%s.raw", extraDiskPath)
-			qcow2Path := fmt.Sprintf("%s.qcow2", extraDiskPath)
-			if err = imgutil.QCOWToRaw(extraDiskPath, rawPath); err != nil {
-				return fmt.Errorf("failed to convert qcow2 disk %q to raw for vz driver: %w", diskName, err)
+			oldFormatPath := fmt.Sprintf("%s.%s", extraDiskPath, extraDiskFormat)
+			if err = imgutil.ConvertToRaw(extraDiskPath, rawPath); err != nil {
```

Comment:
Make sure that `extraDiskFormat` does not contain `os.PathSeparator`

Reply:
Thanks, updated in latest rev